include/toolkit/cache/lib.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright © 2020 Amazon.com, Inc. or its affiliates.
+ * All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef KTF_TOOLKIT_CACHE_H
+#define KTF_TOOLKIT_CACHE_H
+
+#include <lib.h>
+#include <page.h>
+
+#define CACHE_LINE_SIZE     64
+#define CACHE_CHANNEL_LINES 64
+#define CACHE_LINE1         31 /* cache line offset for trigger function 1 */
+#define CACHE_LINE2         (2 * CACHE_LINE1) /* cache line offset for function 2 */
+
+#define SAMPLES_COUNT 64
+
+typedef enum { H_NULL, H_ONE, H_UNDECIDED, H_EMPTY } heisenbit_t;
```

Comment:
Assuming 
H_NULL means 0
H_ONE means 1
What's the difference between H_UNDECIDED and H_EMPTY?
Is it like 
UNDECIDED means, there was an evaluation and we couldn't determine.
EMPTY means, there was no evaluation and this default state?

Reply:
The `H_UNDECIDED` means: both cache lines have been touched
The 'H_EMPTY' means: none of the cache lines have been touched.